components/payments/content/bat_payment_browsertest.cc

```diff
@@ -0,0 +1,196 @@
+#include "brave/components/brave_rewards/browser/test/common/rewards_browsertest_context_util.h"
+#include "brave/components/brave_rewards/browser/test/common/rewards_browsertest_response.h"
+#include "brave/components/brave_rewards/browser/test/common/rewards_browsertest_util.h"
+#include "chrome/test/base/testing_profile.h"
+#include "chrome/test/base/ui_test_utils.h"
+#include "chrome/test/payments/payment_request_platform_browsertest_base.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/test/browser_test.h"
+#include "content/public/test/browser_test_utils.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+using content::WebContentsConsoleObserver;
+using rewards_browsertest::RewardsBrowserTestContextHelper;
+using rewards_browsertest::RewardsBrowserTestResponse;
+
+namespace payments {
+
+class BatPaymentTest : public PaymentRequestPlatformBrowserTestBase {
+ public:
+  BatPaymentTest() {
+    response_ = std::make_unique<RewardsBrowserTestResponse>();
+  }
+
+  void SetUpOnMainThread() override {
+    InProcessBrowserTest::SetUpOnMainThread();
+
+    context_helper_ =
+        std::make_unique<RewardsBrowserTestContextHelper>(browser());
+
+    // HTTP resolver
+    brave::RegisterPathProvider();
+    base::FilePath test_data_dir;
+    base::PathService::Get(brave::DIR_TEST_DATA, &test_data_dir);
+    test_data_dir = test_data_dir.Append("rewards-data");
+    test_data_dir = test_data_dir.Append("payments");
+	https_server()->ServeFilesFromDirectory(test_data_dir);
+
+    host_resolver()->AddRule("*", "127.0.0.1");
+    https_server()->SetSSLConfig(net::EmbeddedTestServer::CERT_OK);
+    ASSERT_TRUE(https_server()->Start());
+
+    // Rewards service
+    brave::RegisterPathProvider();
+    auto* profile = browser()->profile();
+    rewards_service_ = static_cast<brave_rewards::RewardsServiceImpl*>(
+        brave_rewards::RewardsServiceFactory::GetForProfile(profile));
+
+    // Response mock
+    base::ScopedAllowBlockingForTesting allow_blocking;
+    response_->LoadMocks();
+    rewards_service_->ForTestingSetTestResponseCallback(
+        base::BindRepeating(
+            &BatPaymentTest::GetTestResponse,
+            base::Unretained(this)));
+    rewards_service_->SetLedgerEnvForTesting();
+
+    rewards_browsertest_util::SetOnboardingBypassed(browser());
+
+    // For PaymentRequest
+    test_controller()->SetUpOnMainThread();
+	PlatformBrowserTest::SetUpOnMainThread();
+  }
+
+  void TearDown() override {
+    InProcessBrowserTest::TearDown();
+  }
+
+  void CanMakePaymentWillPass(std::list<TestEvent> event_sequence,
+  	                          std::string expected_result,
+  	                          std::string function) {
+    ResetEventWaiterForEventSequence(event_sequence);
+    ASSERT_TRUE(content::ExecuteScript(GetActiveWebContents(), function));
+    WaitForObservedEvent();
+    ExpectBodyContains(expected_result);
+  }
+
+  void CanMakePaymentWillFail(std::string expected_result,
```

Comment:
Nit: Seems like the parameters to this function and a few others in this file can be const reference.

Reply:
addressed: https://github.com/brave/brave-core/pull/7557/files#diff-3bf39fcf0424a5290d52c433dd5841175d441396839895f8af35cd8426a1b3ffR74